src/linter.js

```diff
@@ -465,78 +447,69 @@ export default class Linter {
     return true;
   }
 
-  scan(deps = {}) {
-    return this.extractMetadata(deps)
-      .then(() => {
-        return this.io.getFiles();
-      })
-      .then((files) => {
-        if (this.config.scanFile) {
-          if (!this.config.scanFile.some((f) => Object.keys(files).includes(f))) {
-            const _files = this.config.scanFile.join(', ');
-            throw new Error(`Selected file(s) not found: ${_files}`);
-          }
+  async scan(deps = {}) {
+    try {
+      await this.extractMetadata(deps);
+      const files = await this.io.getFiles();
+      if (this.config.scanFile) {
+        if (!this.config.scanFile.some((f) => Object.keys(files).includes(f))) {
+          const _files = this.config.scanFile.join(', ');
+          throw new Error(`Selected file(s) not found: ${_files}`);
+        }
+      }
+
+      // Known libraries do not need to be scanned
+      const filesWithoutJSLibraries = Object.keys(files).filter((file) => {
+        return !(file in this.addonMetadata.jsLibs);
+      });
+
+      await this.scanFiles(filesWithoutJSLibraries);
+
+      this.print(deps._console);
+      // This is skipped in the code coverage because the
+      // test runs against un-instrumented code.
+      /* istanbul ignore if  */
```

Comment:
Nit, do you mind to remove one space from the end of this comment? (it was already there, but given that we are touching this function... ;-), same at line 492)